Replace export aliases with import decls.

bsermons opened the pull request:
This is my first stab at fixing #90 and #118.  It's not perfect but it gets us to a little better state I think. 

One thing I know is a problem with this is that this just picks up the first import alias it finds, so an alias mapped to multiple modules would only import the first one.  Probably an easy fix so I figured I'd try and get any feedback before continuing.  Looking for any suggestions for improvement also as I'm pretty new to Haskell.

bsermons committed:
Replace export aliases with import decls.

kRITZCREEK commented:
Thanks for putting in all this work – I'll look at it right away. From a quick scan it looks really nice!

kritzcreek committed:
cosmetic changes

kRITZCREEK commented:
So I tried it on a minimal testcase and it seems to work nicely :+1:
I've put together some cosmetic changes here https://github.com/bsermons/psc-ide/pull/1. If there is any questions feel free to ask.

Could you maybe add a testcase or two to ReexportsSpec.hs?

I have an idea on how to extend this to the case of multiple modules mapped to the same alias, but I don't want to spoil you the fun! :smile: If you get stuck we can work on it together.

Thank you!

bsermons committed:
Merge pull request #1 from kRITZCREEK/master-bsermons

cosmetic changes

Commit:
Handle multiple alias imports for reexport

Commit:
Add unit test for aliased reexports

bsermons commented:
How does this look?  This now handles multiple aliased reexports and added a unit test.

Let me know if you can think of any other test cases that might be useful.  I just added one to test that multiple aliases get replaced.
